nilearn/connectome/connectivity_matrices.py

```diff
@@ -390,4 +492,73 @@ def transform(self, X):
                                  '"covariance" and "precision", got kind '
                                  '"{}"'.format(self.kind))
 
-        return np.array(connectivities)
+        connectivities = np.array(connectivities)
+        if self.kind != 'tangent':
+            self.mean_ = connectivities.mean(axis=0)
+
+        if self.vectorize:
+            connectivities = sym_matrix_to_vec(
+                connectivities, discard_diagonal=self.discard_diagonal)
+
+        return connectivities
+
+    def _check_fitted(self):
+        if not hasattr(self, "cov_estimator_"):
+            raise ValueError('It seems that {0} has not been fitted. '
+                             'You must call fit() before calling '
+                             'transform().'.format(self.__class__.__name__)
+                             )
+
+    def inverse_transform(self, connectivities):
+        """Returns connectivity matrices from vectorized or not connectivities.
+
+        If kind is 'tangent', the covariance matrices are reconstructed.
+
+        Parameters
+        ----------
+        connectivities : list of n_subjects numpy.ndarray with shapes\
+            (n_features, n_features) or (n_features * (n_features + 1) / 2,)
+            Connectivities of each subject, vectorized or not.
+
+        Returns
+        -------
+        output : numpy.ndarray, shape (n_subjects, n_features, n_features)
+            The corresponding connectivity matrices. If kind is 'correlation'/
+            'partial correlation', the correlation/partial correlation
+            matrices are returned.
+            If kind is 'tangent', the covariance matrices are reconstructed.
+        """
+        self._check_fitted()
+
+        connectivities = np.array(connectivities)
+        if self.vectorize:
+            if self.discard_diagonal:
+                diagonals_shape = (connectivities.shape[0],
+                                   self.mean_.shape[0])
+                if self.kind in ['correlation', 'partial correlation']:
+                    diagonal = np.ones(diagonals_shape) / sqrt(2.)
+                elif self.kind in ['covariance', 'precision']:
+                    warnings.warn("diagonal values have been discared and are "
```

Comment:
Can you check typos in this warning message ? discared -> discarded unkown-> unknown